app/public/[id]/page.tsx

```diff
@@ -42,6 +45,36 @@ export default function Listing({ params }: { params: { id: string } }) {
         } else {
           setListingData(data as ListingData);
         }
+        const listing = data as ListingData;
+
+        // Check if the retrieved data is a valid ListingData
+        if ("deadline" in data) {
+          // Check if the deadline is a valid date
+          const deadline = new Date(listing.deadline);
+          if (!isNaN(deadline.getTime())) {
+            const now = new Date();
+            const estNow = new Date(now.toLocaleString("en-US", { timeZone: "America/New_York" }));
+            
+            if (estNow > deadline) {
+              console.error("Deadline has passed.");
+              router.push("/error");
```

Comment:
From the applicant's perspective- they won't see the console. There should be a separate error page to handle this. Same goes for other types of errors.

Reply:
I think getting rid of `console.error` is a good idea – however, not a huge fan of showing the details on why a user shouldn't be able to access a listing since it gives too much unnecessary info to them! How about we'll leave it like this for now and I'll make the changes on the logs? @cwgough